feat: add enhancement template (#2486)
dexidp/dex: enhancements/README.md

> # Enhancement Proposal

Reviewer: We should add a link to this file in https://github.com/dexidp/dex/blob/master/.github/ISSUE_TEMPLATE/config.yml so users can find it when looking to open an issue.

Reviewer: Also:
- Should we call these proposals DEPs (Dex Enhancement Proposals?)
- Should we setup a separate repository for them?
- Should we move the whole thing under `docs/` or some other more generic directory?
- How would EPs related to ADRs? Should we use both?

Author: Speaking about the issue template - no problem, I will add a link.

1. I'm good with it. It is more identical, and we can differentiate proposals for different projects in the future.
2. I think it would be best to start with this repo because now we do not have too many cases for writing proposals.  If it becomes inconvenient, we will always be able to move proposals to a separate repo.
3. This is a tricky question. As I see it, proposals are not connected to anything generic. That is why I moved them to the root. Although, it is discussable. If you think we should move them to docs, it is okay for me.
4. I assume we will have EPs for all complicated architecture choices in the future, so it is not necessary to use both.

## Why do we need it?

Enhancement Proposal (EP) is a design document providing information to the community, or describing a new feature for Dex.

> We intend EPs to be the primary mechanisms for proposing major new features.

Reviewer: ```suggestion
We intend EPs to be the primary mechanisms for proposing major new features or significant changes to existing ones.
```

> This will make it easier for community to describe, track, and look through the history of changes that a going to affect the future development of a product.

Reviewer: ```suggestion
This will make it easier for the community to describe, track, and look through the history of changes that affected the development of the project.
```

## Process

### Before starting

> 1. Search GitHub for [previous issues](https://github.com/dexidp/dex/issues) and [EPs](https://github.com/dexidp/dex/tree/master/enhancements).

Reviewer: ```suggestion
1. Search GitHub for previous [issues](https://github.com/dexidp/dex/issues), [discussions](https://github.com/dexidp/dex/discussions) and [EPs](https://github.com/dexidp/dex/tree/master/enhancements).
```

> 2. If an issue does not exist, [open it](https://github.com/dexidp/dex/issues/new?assignees=&labels=&template=feature_request.yaml).

Reviewer: Maybe we should use discussions instead?
